airflow/providers/microsoft/azure/hooks/powerbi.py

```diff
@@ -0,0 +1,323 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import time
+from enum import Enum
+from typing import Any, Callable
+
+import requests
+from azure.identity import ClientSecretCredential
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class PowerBIDatasetRefreshFields(Enum):
+    """Power BI refresh dataset details."""
+
+    REQUEST_ID = "request_id"
+    STATUS = "status"
+    END_TIME = "end_time"
+    ERROR = "error"
+
+
+class PowerBIDatasetRefreshStatus:
+    """Power BI refresh dataset statuses."""
+
+    # If the completion state is unknown or a refresh is in progress.
+    IN_PROGRESS = "In Progress"
+    FAILED = "Failed"
+    COMPLETED = "Completed"
+    DISABLED = "Disabled"
+
+    TERMINAL_STATUSES = {FAILED, COMPLETED}
+
+
+class PowerBIDatasetRefreshException(AirflowException):
+    """An exception that indicates a dataset refresh failed to complete."""
+
+
+class PowerBIHook(BaseHook):
```

Comment:
This hook should extend or delegate it's calls to the KiotaRequestAdapterHook, that way you will be able to use the async code and also not re-invent the wheel once over again